### ISSUE TITLE
Extend NvimTreeOpen / NvimTreeToggle to focus tree to specific file/folder

### ISSUE THREAD

**Issue by hay-kot** [nvim-tree/nvim-tree.lua#2184]
**Can this functionality be implemented utilising API?**

It's possible to extend the API to accept another boolean parameter to choose this type of functionality.

**Is your feature request related to a problem? Please describe.**

This feature would help facilitate my workflow of creating a new file or folder. Often times I'm working on feature "X" and I pivot to build feature "Y" and that requires me to navigate to another part of the project, create a new file/folders and start working in that area. Currently the fastest way have to do this is to click-through/navigate the tree to where I need to be, or fuzzy find a file, open it, and rely on the behavior of nvim-tree to open the current buffer location in the tree.

**Describe the solution you'd like**

I would like to be able to provide a argument to NvimTreeOpen, NvimTreeToggle or API's to open the standard tree and focus on a specific file or directory. This would allow me to use telescope to fuzzy find the directory, pass that to nvim-tree and then create files/folders as necessary.

Basically, instead of focusing on the current buffer, focus on the path I provided. 

**Describe alternatives you've considered**

I know I can pass a file path directly, but if I pass the folder it sets the root context to the directory, which is _not_ what I want because on subsequence toggles the root context remains at the nested folder and not the project root.

**Comment by alex-courtis** [comment]
That can work.

`{find_file}` would need to be implicitly set to `false`

`{update_root}` or `update_focused_file.update_root` and friends would be used to control whether the tree root changes.

**Comment by hay-kot** [comment]
Actually, as I was looking through the codebase to make a PR and I found `api.tree.find_file({path})` which is exactly what I needed. For anyone else trying to accomplish the same thing here's a snipped using telescope to focus on a specific directory. 

```lua
        function()
          local actions = require("telescope.actions")
          local action_state = require("telescope.actions.state")

          local function open_nvim_tree(prompt_bufnr, _)
            actions.select_default:replace(function()
              local api = require("nvim-tree.api")

              actions.close(prompt_bufnr)
              local selection = action_state.get_selected_entry()
              api.tree.open()
              api.tree.find_file(selection.cwd .. "/" .. selection.value)
            end)
            return true
          end

          require("telescope.builtin").find_files({
            find_command = { "fd", "--type", "directory", "--hidden", "--exclude", ".git/*" },
            attach_mappings = open_nvim_tree,
          })
        end,
```

Closing since this can be accomplished with existing API's.

**Comment by alex-courtis** [comment]
> Actually, as I was looking through the codebase to make a PR and I found `api.tree.find_file({path})` which is exactly what I needed.

Ah, good catch!

**Comment by alex-courtis** [comment]
> For anyone else trying to accomplish the same thing here's a snipped using telescope to focus on a specific directory.

Can you please add a [Recipe](https://github.com/nvim-tree/nvim-tree.lua/wiki/Recipes) ? This is really useful.

**Comment by hay-kot** [comment]
Done!

https://github.com/nvim-tree/nvim-tree.lua/wiki/Recipes#find-and-focus-directory-with-telescope
